examples/flutter_gallery/lib/demo/pan_and_zoom_demo_board.dart

```diff
@@ -0,0 +1,281 @@
+import 'dart:collection' show IterableMixin;
+import 'dart:math';
+import 'dart:ui' show Vertices;
+import 'package:flutter/material.dart' hide Gradient;
+import 'package:vector_math/vector_math_64.dart' show Vector3;
+
+// An abstraction of the hex board logic
+class Board extends Object with IterableMixin<BoardPoint> {
```

Comment:
@HansMuller I use `Board` as an immutable, so changing anything results in a new instance.  Following that pattern, I only need to do a shallow comparison of the memory location to check equality.

Is that the kind of comparison that will happen if I don't override `==`?  And overall is that a reasonable pattern to use in an example like this?  I don't want to confuse people reading this code if it's really unusual, so if so I'll change it and override `==`.

Reply:
Currently `Border(selected: false, boardPoints: []) != Border(selected: false, boardPoints: [])`.   If that's OK for this app then the overrides aren't needed. Providing overrides for hashCode and == isn't that common; I was just concerned because of the == in BoardPainter.shouldRepaint().

Reply:
Ah good point.  I'm going to try this: in my methods that return a new `Board`, if the result would be an instance with the same values, then I'll just return the original instance.  I think that will prevent my worries about comparing the same instance in `shouldRepaint`.